### DISCUSSION TITLE
File-related methods

### DISCUSSION THREAD

**Discussion by Poshi** [johnkerl/miller#1558] [Ideas]
It's me the only one that lacks the ability of working with files in the data processed?
From time to time, my CSV files contain the paths to files. Sometimes I'd like to do some processing like: if the file does not exist, remove the record; or put in a new column the file size; or keep only the files newer than one week.
A set of DSL methods to deal with them would be useful to me. Existence test, type test, size and dates stats...
Anyone else with these needs?

**Comment by johnkerl** [comment]
@Poshi would the list include the following?

* file size
* file type (file, directory, symlink, fifo, etc)
* exists
* create/modify time

In short, the `stat` system call -- perhaps returned as a Miller map?

* https://pkg.go.dev/os#File.Stat
* https://pkg.go.dev/io/fs#FileInfo

**Comment by Poshi** [comment]
Yes, I'm mostly thinking on the data that a `stat` call can return. I'm not intending to access or modify the contents of the files in any way, but working on its metadata from a read only perspective (I'm not trying to mimmick `touch`).
Having the contents of a stat call would avoid having to create methods for all the possible tests, as this would be done by `stat`-ing the file and then applying the filters on that structure. That seems a concise solution :-)
Right now, once I groomed the data, I have to parse the resulting file from bash to make the tests and regenerate the file accordingly. It would be a lot easier if these filters can be applied from Miller itself.
